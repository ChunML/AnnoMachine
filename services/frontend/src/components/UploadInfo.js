import React from 'react';
import PropTypes from 'prop-types';

function UploadInfo({ username, uploaded_at }) {
  return (
    <div className="segment">
      <p>
        Author:{' '}
        <span style={{ color: 'tomato', fontSize: '1.6rem' }}>{username}</span>
      </p>
      <p>
        Uploaded at:{' '}
        <span style={{ color: 'chocolate', fontStyle: 'italic' }}>
          {uploaded_at}
        </span>
      </p>
    </div>
  );
}

UploadInfo.propTypes = {
  username: PropTypes.string.isRequired,
  uploaded_at: PropTypes.string.isRequired,
};

export default UploadInfo;
