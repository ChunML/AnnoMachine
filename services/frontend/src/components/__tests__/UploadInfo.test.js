import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import UploadInfo from '../UploadInfo';
import '../../setupTests';

const component = (
  <UploadInfo username="testUser" uploaded_at="2000/01/01 01:02" />
);

it('UploadInfo renders properly', () => {
  const wrapper = shallow(component);
  const info = wrapper.find('p');
  expect(info.length).toBe(2);
  expect(info.at(0).text()).toEqual('Author: testUser');
  expect(info.at(1).text()).toEqual('Uploaded at: 2000/01/01 01:02');
});

it('UploadInfo renders a snapshot properly', () => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
