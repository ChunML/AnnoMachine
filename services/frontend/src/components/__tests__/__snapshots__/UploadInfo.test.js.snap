// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadInfo renders a snapshot properly 1`] = `
<div
  className="ui left aligned red segment"
>
  <p>
    Author: 
    <span
      style={
        Object {
          "color": "tomato",
          "fontSize": "1.6rem",
        }
      }
    >
      testUser
    </span>
  </p>
  <p>
    Uploaded at: 
    <span
      style={
        Object {
          "color": "chocolate",
          "fontStyle": "italic",
        }
      }
    >
      2000/01/01 01:02
    </span>
  </p>
</div>
`;
